rust/nft-candy-machine/src/lib.rs

```diff
@@ -27,7 +27,33 @@ pub mod nft_candy_machine {
     };
 
     use super::*;
+    pub fn withdraw_funds<'info>(ctx: Context<WithdrawFunds<'info>>) -> ProgramResult {
+        let authority = &ctx.accounts.authority;
+        let charity = &ctx.accounts.charity;
+        let pay = &ctx.accounts.config.to_account_info();
+        let snapshot: u64 = pay.lamports();
+        let auth_pot = snapshot
+            .checked_div(10)
+            .ok_or(ErrorCode::NumericalOverflowError)?
```

Comment:
As cool as this is, the charity needs to be on the JS side. Should be optional not required :)

Reply:
fine but you're askin for trouble

people have demanded I confiscate some $ for dev purposes

it won't be 10%

it will be hard-coded on contract :) this code does not go to waste

Reply:
also the idea was to override the charity address

to urs

if u hate charity

Reply:
@bhgames @austbot in hindsight that weird sol transfer becomes unbalanced tx when I try shit that isn't a magnitude of 10

so it's also js